internal/mode/static/state/graph/gateway_listener.go

```diff
@@ -17,6 +17,7 @@ import (
 // Listener represents a Listener of the Gateway resource.
 // For now, we only support HTTP and HTTPS listeners.
 type Listener struct {
+	Name string
```

Comment:
We can derive `Name` from `Source`. I currently add it because a lot of tests do not specify `Source`. Do you have any preferences?

Reply:
I like this approach. It makes the tests easier to write.